src/providers/ldap/sdap.c

```diff
@@ -1254,7 +1254,8 @@ errno_t sdap_set_config_options_with_rootdse(struct sysdb_attrs *rootdse,
     int ret;
     char *naming_context = NULL;
 
-    if (!sdom->search_bases
+    if (!sdom->naming_context
```

Comment:
If any of this variables is `NULL` we will call `get_naming_context()` to retrieve the naming context. If `sdom->naming_context` happens to not be `NULL`, wouldn't it be simpler to use it now we are keeping the `naming_context` in `sdom` and avoid using the intermediate `naming_context` variable with its memory allocation, release and a call to `talloc_strdup()`?

We could simply check for `sdom->naming_context` being NULL and only in that case, call `get_naming_context()` . From there, we can use `sdom->naming_context`.

Something like this:
```
if (!sdom->naming_context) {
    sdom->naming_context = get_naming_context(sdom, rootdse);
    if (sdom->naming_context == NULL) {
        DEBUG(...);
        ret = EOK;
        goto done;
    }
}
```
(From here start using sdom->naming_context instead of naming_context)
```
/* Default */
if (!sdom->search_bases) {
    ret = sdap_set_search_base(opts, sdom,
                               SDAP_SEARCH_BASE,
                               sdom->naming_context);
    if (ret != EOK) goto done;
}
...
```